### PR TITLE
[Customer Center] Build `WrongPlatformView` from JSON

### PR DESCRIPTION
**Pull request by vegaro**
Build `WrongPlatformView` using Strings from the JSON

Cleaned up the view as well so the logic is simpler and looks cleaner

Simulator Screenshot - iPhone 15 Pro - 2024-08-30 at 18 43 15

### CODE REVIEW

File: Sources/CustomerCenter/CustomerCenterConfigData.swift [hunk 1]
```diff
@@ -133,6 +138,16 @@ public struct CustomerCenterConfigData {
                     return "Update"
                 case .updateWarningIgnore:
                     return "Continue"
+                case .platformMismatch:
+                    return "Platform mismatch"
```
Comment (vegaro) [inline Sources/CustomerCenter/CustomerCenterConfigData.swift:142]: Happy to talk about this. I went through different ideas, even not showing a title at all. Let me know what you think

image

File: Sources/CustomerCenter/CustomerCenterConfigData.swift [hunk 2]
```diff
@@ -133,6 +138,16 @@ public struct CustomerCenterConfigData {
...
+                case .pleaseContactSupportToManage:
+                    return "Please contact support to manage your subscription."
```
Comment (jamesrb1) [inline Sources/CustomerCenter/CustomerCenterConfigData.swift:144]: Maybe not for this PR, but would be great if there were a link/button that would take them directly to contacting support from this page.

File: Sources/CustomerCenter/CustomerCenterConfigData.swift [hunk 3]
```diff
@@ -133,6 +138,16 @@ public struct CustomerCenterConfigData {
...
+                case .appleSubscriptionManage:
+                    return "You can manage your subscription via the App Store app on an Apple device."
```
Comment (jamesrb1) [inline Sources/CustomerCenter/CustomerCenterConfigData.swift:146]: I would probably replace "via" with "by using" here, but just an opinion. But actually, can you with the app store app? I think this needs to be managed in Settings or via your AppleID somehow.

File: Sources/CustomerCenter/CustomerCenterConfigData.swift (hunk 2 above)
Comment (tonidero) [reply Sources/CustomerCenter/CustomerCenterConfigData.swift:144]: That button is probably something you can configure from the JSON :)

**Review by tonidero — APPROVED** [review]
LGTM!

File: Sources/CustomerCenter/CustomerCenterConfigData.swift (hunk 3 above)
Comment (vegaro) [reply Sources/CustomerCenter/CustomerCenterConfigData.swift:146]: Thanks for the feedback, I will update the text. You can actually manage the subscriptions in the app store by clicking the profile pic

Comment (jamesrb1) [reply Sources/CustomerCenter/CustomerCenterConfigData.swift]: > You can actually manage the subscriptions in the app store by clicking the profile pic

TIL!

**Review by jamesrb1 — APPROVED** [review]

**Pull request merged**